.github/workflows/clippy.yml

```diff
@@ -0,0 +1,42 @@
+##
+## Perform Clippy checks - currently set to defaults
+##   https://github.com/rust-lang/rust-clippy#usage
+##   https://rust-lang.github.io/rust-clippy/master/index.html
+##
+
+name: Clippy Checks
+
+# Only run when:
+#   - PRs are (re)opened against develop branch
```

Comment:
Should probably also run against the `next` and `master` branches

Reply:
Can do - would like to hear what @igorsyl thinks about this as well, and if we should be adding in some exclusions to what we want to be warned about.

Reply:
We can enable the least number of categories to produce a few suggestions and later add more categories as we fix issues. Is there a way to auto-generate issues from a single clippy run?

```
Category | Description | Default level
-- | -- | --
clippy::all | all lints that are on by default (correctness, suspicious, style, complexity, perf) | warn/deny
clippy::correctness | code that is outright wrong or useless | deny
clippy::suspicious | code that is most likely wrong or useless | warn
clippy::style | code that should be written in a more idiomatic way | warn
clippy::complexity | code that does something simple but in a complex way | warn
clippy::perf | code that can be written to run faster | warn
clippy::pedantic | lints which are rather strict or have occasional false positives | allow
clippy::nursery | new lints that are still under development | allow
clippy::cargo | lints for the cargo manifest | allow
```

https://github.com/rust-lang/rust-clippy

Reply:
this is now disabled from running until we can address/test how chatty this will be